compiler/ast.nim

```diff
@@ -571,8 +571,8 @@ type
   TMagic* = enum # symbols that require compiler magic:
     mNone,
     mDefined, mDefinedInScope, mCompiles, mArrGet, mArrPut, mAsgn,
-    mLow, mHigh, mSizeOf, mTypeTrait, mIs, mOf, mAddr, mTypeOf, mRoof, mPlugin,
-    mEcho, mShallowCopy, mSlurp, mStaticExec,
+    mLow, mHigh, mSizeOf, mTypeTrait, mIs, mOf, mAddr, mType, mTypeOf,
```

Comment:
Why is 'mType' required? It can be mapped to 'mTypeOf'?

Reply:
I'll just remove `mTypeOf` later. It's currently needed only for bootstrapping with the older sources.

Reply:
well my question is: Why not keep using mTypeOf for this?

Reply:
Because someday someone will wonder, why is the `type` special type called `TypeOf` in the compiler internals. For this reason, we should also rename `tyTypeDesc` eventually.

Reply:
Well it's not like the compiler isn't full of these things already, but ok. :-)